allennlp/modules/transformer/t5.py

```diff
@@ -326,17 +340,16 @@ def project(hidden_states, proj_layer, key_value_states, past_key_value) -> Floa
                 position_bias = position_bias[:, :, -seq_length:, :]
 
             if mask is not None:
-                position_bias = apply_mask(
-                    position_bias, mask
-                )  # (batch_size, n_heads, seq_length, key_length)
+                # Shape: (batch_size, num_heads, seq_length, key_length)
+                position_bias = position_bias + mask
```

Comment:
I changed this to use our `apply_mask()`, and you changed it back. I assume you looked into this and you know what's going on?

I seem to remember that with masking, HF likes to make a float mask early on and then use that everywhere, whereas I was trying to pass around boolean masks and make them into floats at the last possible moment (if ever). I think my approach is better, because you can use the boolean mask in different ways in different places, for example, apply in an fp16 way in one place, but apply in a fp32 way in another place, or use `sum()` to find the number of non-masked items, etc.

Reply:
Done.